core/src/main/c/share/ooo_radix.h

```diff
@@ -0,0 +1,832 @@
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 8:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<8u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 16:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<16u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 24:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<24u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 32:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<32u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 40:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<40u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 48:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<48u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        case 56:
+            return radix_copy_segments_index_asc_dispatch_txn_segment_bits<56u>(
+                    txn_bits, segment_bits, lag_ts, lag_size,
+                    segment_ts, segment_txns, txn_count, out, cpy,
+                    segment_count, min_value, total_row_count_bytes, total_row_count, result_format);
+        default:
+            return -1;
+    }
+}
+
+template<typename T, uint16_t segment_bits>
+void merge_shuffle_column_from_many_addresses_segment_bits(const T **src_addresses, T *dst_address,
+                                                           const index_l *merge_index_address, uint64_t row_count) {
+    constexpr uint64_t mask = (1ULL << segment_bits) - 1;
```

Comment:
In many places, non-type template parameters can be safely replaced with runtime constants